classes/ETL/Ingestor/HostTableTransformIngestor.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+* This class iterates over a staging table (record_time_staging) in which each row pairs a 
+* host with a hardware configuration at a specific time (record_time). 
+* The transformation will associate each host with a specific hardware configuration
+* over a *range* of times (start_time and end_time)
+* 
+* @author Max Dudek <maxdudek@gmail.com>
+* @date 2019-05-20
+*/
+
+
+namespace ETL\Ingestor;
+
+use ETL\aOptions;
+use ETL\iAction;
+use ETL\aAction;
+use ETL\Configuration\EtlConfiguration;
+use ETL\EtlOverseerOptions;
+
+use Log;
+
+class HostTableTransformIngestor extends pdoIngestor implements iAction
+{
+
+    /**
+     * @var $_instance_state an array representing a row to be added to the host table - 
+     * it associates a host with a hardware configuration during a period of time
+     */
+    private $_instance_state;
+
+    /**
+     * @see ETL\Ingestor\pdoIngestor::__construct()
+     */
+    public function __construct(aOptions $options, EtlConfiguration $etlConfig, Log $logger = null)
+    {
+        parent::__construct($options, $etlConfig, $logger);
+
+        $this->_instance_state = null;
+    }
+
+    /**
+     * Create a new row and associate it with the current record on the staging table
+     * 
+     * @param $srcRecord the current row from the staging table being read
+     */
+    private function initInstance($srcRecord)
```

Comment:
If this is always going to be an array, change the function parameter to `array $srcRecord`, otherwise state the expected type in the @param line of the docblock.